WordPress/gutenberg
# Add new `HeadingLevel` icons

Part of ##50520;
Related to: #46003, #50402

## What?

This PR adds the following six icons to the `icons` package:

- `HeadingLevel1`
- `HeadingLevel2`
- `HeadingLevel3`
- `HeadingLevel4`
- `HeadingLevel5`
- `HeadingLevel6`

## Why?

As noted in [this comment](https://github.com/WordPress/gutenberg/issues/50520#issue-1704108555), the icons representing the heading level are used in the Heading block and the Site Title block. These icons are hard coded and not optimized for size 24.

Since these icons already exist in [the Figma library](https://www.figma.com/file/e4tLacmlPuZV47l7901FEs/WordPress-Design-Library?type=design&node-id=8%3A13&t=9HJwOHgKLwXYRnZn-1), it would make sense to add this to the `icons` package and apply it to these blocks.

## How?

I used [this SVG data](https://github.com/WordPress/gutenberg/issues/50520#issuecomment-1556800845) provided by @jasmussen 🙏

## Testing Instructions

- Run `npm run storybook:dev`
- Access http://localhost:50240/?path=/story/icons-icon--library
- Confirm that six heading icons are added.

![storybook](https://github.com/WordPress/gutenberg/assets/54422211/6ebcff2f-9279-4bc4-8e6b-65b341aed1e8)

- Use the following code to update the edit component of the code block (`packages/block-library/src/code/edit.js`).

<details>
<summary>Code</summary>

```javascript
/**
 * WordPress dependencies
 */
import { useBlockProps } from '@wordpress/block-editor';
import { Button, Flex } from '@wordpress/components';
import {
	Icon,
	headingLevel1,
	headingLevel2,
	headingLevel3,
	headingLevel4,
	headingLevel5,
	headingLevel6,
} from '@wordpress/icons';

const headingIcons = {
	1: headingLevel1,
	2: headingLevel2,
	3: headingLevel3,
	4: headingLevel4,
	5: headingLevel5,
	6: headingLevel6,
};

export default function CodeEdit() {
	return (
		<div { ...useBlockProps() }>
			<Flex direction="column">
				{ [ 1, 2, 3, 4, 5, 6 ].map( ( level ) => (
					<Flex key={ level } justify="flex-start" wrap={ true }>
						<Icon icon={ headingIcons[ level ] } />
						<Button
							icon={ headingIcons[ level ] }
							variant="primary"
						>
							{ `Heading ${ level }` }
						</Button>
						<Button
							icon={ headingIcons[ level ] }
							variant="secondary"
						>
							{ `Heading ${ level }` }
						</Button>
						<Button
							icon={ headingIcons[ level ] }
							variant="tertiary"
						>
							{ `Heading ${ level }` }
						</Button>
						<Button icon={ headingIcons[ level ] } isPressed>
							{ `Heading ${ level }` }
						</Button>
					</Flex>
				) ) }
			</Flex>
		</div>
	);
}
```

</details> 

- Insert the code block in the block editor and confirm that all icons are properly displayed.

![block-editor](https://github.com/WordPress/gutenberg/assets/54422211/9323abb6-5921-4dae-95b2-342c31e6463b)

## Related Issue/PR

In the heading block where these icons are used, there is an associated issue/PR:

- #50402
- #46003 (Submitted by @carolinan)

Once this PR is merged, I believe that these icons can also be used in #46003 to remove the hard-coded heading icon data.

I think we need to continue to discuss the heading dropdown UI changes that are on the agenda for #50402.
